src/js/select2/core.js

```diff
@@ -327,6 +327,18 @@ define([
           self.open();
 
           evt.preventDefault();
+        } else if (key === KEYS.DOWN || key === KEYS.UP) {
+            var options = this.$element.find('option:selected');
+            options = key === KEYS.DOWN ? 
+                options.nextAll(':enabled') :
+                options.prevAll(':enabled');
+
+            var val = options.first.val();
```

Comment:
You might not like to change the selected multiple items.
That would make line 330:
} else if ((key === KEYS.DOWN || key === KEYS.UP) && (!(self.options.get('multiple'))) ) {